File: app/src/main/java/mastermind/Gamestate.java
package mastermind;

public class Gamestate {
    private final int[][] placedPegs;
    private final boolean repeatable;

    public Gamestate(String saveState, String rep){
        String[] boxStates = saveState.split(" ");
        placedPegs = new int[][]{{6,6,6,6},{6,6,6,6},{6,6,6,6},{6,6,6,6},
                {6,6,6,6},{6,6,6,6},{6,6,6,6},{6,6,6,6},{6,6,6,6}};
        if (boxStates.length == 9){
            for (int i = 0; i < boxStates.length; i++){
                for (int j = 0; j < boxStates[i].length(); j++){
                    placedPegs[i][j] = Character.getNumericValue(boxStates[i].charAt(j));
                }
            }
        }

        repeatable = "true".equals(rep);
    }


    public int[][] getPlacedPegs() {
        return placedPegs;
    }

    public boolean isRepeatable() {
        return repeatable;
    }


}

File: app/src/main/kotlin/mastermind/Gamestate.kt
package mastermind

class Gamestate(saveState: String, rep: String) {
    val placedPegs: Array<IntArray>
    val isRepeatable: Boolean

    init {
        val boxStates = saveState.split(" ").toTypedArray()
        placedPegs = arrayOf(intArrayOf(6, 6, 6, 6), intArrayOf(6, 6, 6, 6), intArrayOf(6, 6, 6, 6), intArrayOf(6, 6, 6, 6), intArrayOf(6, 6, 6, 6), intArrayOf(6, 6, 6, 6), intArrayOf(6, 6, 6, 6), intArrayOf(6, 6, 6, 6), intArrayOf(6, 6, 6, 6))
        if (boxStates.size == 9) {
            for (i in boxStates.indices) {
                for (j in 0 until boxStates[i].length) {
                    placedPegs[i][j] = Character.getNumericValue(boxStates[i][j])
                }
            }
        }
        isRepeatable = "true" == rep
    }
}